simple_vision_demo: Upload results to Superset for all batch sizes

pbaraTT opened the pull request:
### Ticket
https://github.com/tenstorrent/tt-metal/issues/21007

### Problem description
We want to upload performance results to Superset for Llama3.2-11B-Vision for every batch size: {1, 4, 16, 32}. We also want to verify performance (with assert) for this model for batch size 16.

### What's changed
+ simple_vision_demo.py: Upload results to Superset for all batch sizes
+ introduce case for batch size = 16
+ limit performance assertion to the following test cases:
  - N300 + Llama-3.2-11B + batch 16
  - T3K + Llama-3.2-90B + batch 1
+ In order to re-enable Llama-3.2-11B tests with N300 mesh on T3000, I effectively reverted PR [#28666](https://github.com/tenstorrent/tt-metal/pull/28248).

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
  - https://github.com/tenstorrent/tt-metal/actions/runs/17841879995
- [x] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
  - https://github.com/tenstorrent/tt-metal/actions/runs/17830393570/job/50694897772

pbaraTT committed:
simple_vision_demo: Upload results to Superset for all batch sizes: {1, 4, 16, 32}

+ introduce case for batch size = 16
+ limit performance assertion to the following test cases:
  - N300 + Llama-3.2-11B + batch 16
  - T3K + Llama-3.2-90B + batch 1

pbaraTT committed:
t3000_demo_tests CI WF: add batch16 to tests

+ increase timeout to 900 secs to accomodate the additional test

pbaraTT committed:
t3000_demo_tests: add N300 back to tests

- we need N300 runs to report perf to Superset

pbaraTT committed:
[TO BE REVERTED] - CI: only run  Llama3.2 11B vision tests in t3000 demo

skhorasganiTT commented on models/tt_transformers/demo/simple_vision_demo.py:
```diff
@@ -420,47 +420,47 @@ def test_multimodal_demo_text(
         }
 
+        perf_targets = {}
+        if run_config in targets_prefill_tok_s:
+            perf_targets = {
+                "prefill_t/s": targets_prefill_tok_s[run_config],
+                "decode_t/s": targets_decode_tok_s_u[run_config] * max_batch_size,
+                "decode_t/s/u": targets_decode_tok_s_u[run_config],
+            }
```
Should assert that the target also exists for decode.

skhorasganiTT commented on models/tt_transformers/demo/simple_vision_demo.py:
```diff
@@ -420,47 +420,47 @@ def test_multimodal_demo_text(
         # Save benchmark data for CI
         N_warmup_iter = {"inference_prefill": 0, "inference_decode": 0}
-        benchmark_data = create_benchmark_data(profiler, measurements, N_warmup_iter, targets)
+        benchmark_data = create_benchmark_data(profiler, measurements, N_warmup_iter, perf_targets)
```
The intention for the targets uploaded to Superset was to be the targets we aim for (the ones on the GH readmes) rather than the perf targets we assert on, but since this was the existing behaviour in this test, it's ok for now.

skhorasganiTT commented on tests/scripts/t3000/run_t3000_demo_tests.sh:
```diff
@@ -165,8 +165,9 @@ run_t3000_llama3_vision_tests() {
-    MESH_DEVICE=$mesh_device LLAMA_DIR=$llama11b pytest -n auto models/tt_transformers/demo/simple_vision_demo.py -k "batch1-trace or batch4-trace-with-text-prompts" --timeout 600; fail+=$?
+  for mesh_device in "$n300" "$t3k"; do
+    MESH_DEVICE=$mesh_device LLAMA_DIR=$llama11b pytest -n auto models/tt_transformers/demo/simple_vision_demo.py \
+    -k "batch1-trace or batch4-trace-with-text-prompts or batch-16trace" --timeout 900; fail+=$?
```
As discussed offline, pytest skips to be used instead of -k "..."  for testing proper batch sizes.

pbaraTT committed:
Introduce explicit check for wanted batch+mesh configs

- For N300 (2 devices), tested BS={1, 4, 16}
- For T3K (8 devices), tested BS={1, 4, 32}
+ Added assert on target numbers existing for both prefill and decode

pbaraTT committed:
t3000 llama3.2-11B-vision tests: Do not specify cases at invocation

- The control over which batch sizes are tested for given mesh devices is transferred to the test itself

pbaraTT commented on models/tt_transformers/demo/simple_vision_demo.py:
```diff
@@ -420,47 +426,51 @@ def test_multimodal_demo_text(
 
+        perf_targets = {}
+        if run_config in targets_prefill_tok_s:
+            assert (
+                run_config in targets_decode_tok_s_u
+            ), f"Prefill targets exist, but decode targets are missing for {run_config}"
```
The comment is not helpful - it suggests I do exactly what's already there

pbaraTT commented on models/tt_transformers/demo/simple_vision_demo.py:
```diff
@@ -177,15 +178,20 @@ def test_multimodal_demo_text(
-    # Start profiler
-    logger.info(f"Start profiler")
+    num_devices = mesh_device.get_num_devices() if isinstance(mesh_device, ttnn.MeshDevice) else 1
+
+    if num_devices == 2 and max_batch_size not in (1, 4, 16):
+        pytest.skip(f"Batch size={max_batch_size} is not tested for N300 mesh")
+    if num_devices == 8 and max_batch_size not in (1, 4, 32):
+        pytest.skip(f"Batch size={max_batch_size} is not tested for T3K mesh")
+
+    logger.info("Start profiler")
     profiler = BenchmarkProfiler()
     profiler.start("run")
 
     ckpt_dir = os.environ["LLAMA_DIR"]
     tokenizer_path = str(Path(ckpt_dir) / "tokenizer.model")
 
-    num_devices = mesh_device.get_num_devices() if isinstance(mesh_device, ttnn.MeshDevice) else 1
     max_batch_size *= data_parallel  # input batch_size is interpreted as size per DP group
```
Untrue!

skhorasganiTT reviewed (approved):
Thanks @pbaraTT! Could you also post the t3k demo pipeline link in the PR description?

pbaraTT committed:
Skip test for batch1 on N300 mesh due to ND hangs in CI

pbaraTT committed:
remove unused argument for prepare_generator_args()

pbaraTT committed:
llama3.2-11B Vision t3000_demo_tests:  Exclude batch1 without trace test

pbaraTT committed:
revert 300a2983

gwangTT reviewed (approved):
LGTM

pbaraTT committed:
[TO BE REVERTED] CI: enable only llama3.2-11B vision tests again and generate caches

pbaraTT committed:
Revert 0670466b4

kkabilarTT commented on .github/workflows/t3000-demo-tests-impl.yaml:
```diff
@@ -23,24 +23,24 @@ jobs:
+          # # This requirements and comment removed when https://github.com/tenstorrent/tt-metal/pull/22608 merges.
+          # { name: "t3k_qwen3_tests", arch: wormhole_b0, cmd: run_t3000_qwen3_tests, timeout: 60, owner_id: U03HY7MK4BT}, # Mark O'Connor
+          # { name: "t3k_qwen25_vl_tests", arch: wormhole_b0, cmd: run_t3000_qwen25_vl_tests, timeout: 60, owner_id: U07RY6B5FLJ},  #Gongyu Wang
+          # { name: "t3k_gemma3_tests", arch: wormhole_b0, cmd: run_t3000_gemma3_tests, timeout: 30, owner_id: U08TJ70UFRT},  # Harry Andrews
```
Please uncomment these tests

pbaraTT replied:
Apologies. I wanted to re-run the single test with write permissions on `/mnt/MLPerf` so that caches are regenerated. Uncommented
